File: app/src/main/java/com/VegaSolutions/lpptransit/lppapi/responseobjects/ArrivalOnRoute.java
package com.VegaSolutions.lpptransit.lppapi.responseobjects;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class ArrivalOnRoute extends StationOnRoute {

    List<Arrival> arrivals;

    public List<Arrival> getArrivals() {
        return arrivals;
    }

    public class Arrival {

        @SerializedName("route_id")
        private String routeId;

        @SerializedName("vehicle_id")
        private String vehicleId;

        private int type;

        @SerializedName("eta_min")
        private int etaMin;

        @SerializedName("route_name")
        private String routeName;

        @SerializedName("trip_name")
        private String tripName;

        private int depot;

        public String getRouteId() {
            return routeId;
        }

        public String getVehicleId() {
            return vehicleId;
        }

        public int getType() {
            return type;
        }

        public int getEtaMin() {
            return etaMin;
        }

        public String getRouteName() {
            return routeName;
        }

        public String getTripName() {
            return tripName;
        }

        public int getDepot() {
            return depot;
        }

    }

}

File: app/src/main/java/com/VegaSolutions/lpptransit/lppapi/data/ArrivalOnRoute.kt
package com.VegaSolutions.lpptransit.lppapi.data

import com.google.gson.annotations.SerializedName

data class ArrivalOnRoute(
    @SerializedName("name")
    val name: String,

    @SerializedName("station_code")
    val codeId: Int,

    @SerializedName("order_no")
    val orderNo: Int,

    @SerializedName("latitude")
    val latitude: Double,

    @SerializedName("longitude")
    val longitude: Double,

    @SerializedName("station_int_id")
    val stationIntId: Int,

    @SerializedName("arrivals")
    val arrivals: List<Arrival>,
) {
    data class Arrival(
        @SerializedName("route_id")
        val routeId: String,

        @SerializedName("vehicle_id")
        val vehicleId: String,

        @SerializedName("type")
        val type: Int,

        @SerializedName("eta_min")
        val etaMin: Int,

        @SerializedName("route_name")
        val routeName: String,

        @SerializedName("trip_name")
        val tripName: String,

        @SerializedName("depot")
        val depot: Int,
    )
}
